src/geometric_algebra/nursery/chisolm.lean

```diff
@@ -69,12 +82,136 @@ local infix `*₊`:75 := sym_prod
 
 local infix `*₊ᵥ`:75 := sym_prod_vec
 
+
+def is_orthogonal (a : G₁) (b : G₁) : Prop := sym_prod_vec a b = 0
+
+theorem zero_is_orthogonal (a : G₁) : is_orthogonal 0 a := begin
+  unfold is_orthogonal,
+  unfold sym_prod_vec,
+  unfold prod_vec,
+  simp
+end
+
+/- a list of r orthogonal vectors, which may be non-unique -/
+def pairwise_ortho_vector (r : ℕ) := {l : vector G₁ r // l.val.pairwise is_orthogonal}
+
+/- need this for later, can't seem to get the type inference to work if I inline it-/
+-- def zero_pairwise_ortho_vector (r : ℕ) : pairwise_ortho_vector r := ⟨
+--   vector.repeat (0 : G₁) r, begin
+--   unfold vector.repeat subtype.val,
+--   apply pairwise_of_repeat,
+--   apply zero_is_orthogonal,
+-- end⟩
+
+
+-- r-blades
+def is_rblade (r : ℕ) (b : G) : Prop :=
+  -- a product of orthogonal vectors an a scalar
+  (∃ (k: G₀) (v : pairwise_ortho_vector r),
```

Comment:
I was thinking about changing the definition from exists a k to directly say the square of vectors equals the quadratic form. It's always better to say what it is rather than it exists. Although I'm sure about how it extends to other definitions...

Reply:
For grades greater than 1, there is no unique factorization. If possible, I'd rather prove the unique factorization of 0 and 1 blades rather than assume it.